sdks/java/io/bigquery-io-perf-tests/src/test/java/org/apache/beam/sdk/bigqueryioperftests/BigQueryIOIT.java

```diff
@@ -151,46 +172,45 @@ private static void createAndFillSourceTable() {
                       return tableRow;
                     })
                 .withCustomGcsTempLocation(ValueProvider.StaticValueProvider.of(tempRoot))
+                .withMethod(method)
                 .withSchema(
                     new TableSchema()
                         .setFields(
                             Collections.singletonList(
                                 new TableFieldSchema().setName("data").setType("BYTES")))));
-    pipeline.run().waitUntilFinish();
+
+    PipelineResult pipelineResult = pipeline.run();
+    pipelineResult.waitUntilFinish();
+    extractAndPublishTime(pipelineResult, WRITE_TIME_METRIC_NAME);
   }
 
-  private static boolean checkForSourceTable() {
-    BigQueryOptions options = BigQueryOptions.newBuilder().build();
-    BigQuery client = options.getService();
-    TableId tableId =
-        TableId.of(options.getProjectId(), bigQuerySourceDataset, bigQuerySourceTable);
-    Table sourceTable = client.getTable(tableId, TableOption.fields(BigQuery.TableField.TYPE));
-    return sourceTable != null;
+  private void extractAndPublishTime(PipelineResult pipelineResult, String writeTimeMetricName) {
```

Comment:
If we're using this private method in `restRead` too, it should go below it to be more readable. Could you move this?

Reply:
Sure, I rearranged the code. Thanks!